test/tests/functions/piecewise_constant_from_csv/tests

```diff
@@ -50,11 +50,20 @@
       input = 'piecewise_constant.i'
       exodiff = 'piecewise_constant_block.e'
       cli_args = 'ICs/active=block Outputs/file_base=piecewise_constant_block'
-      detail = 'with data sorted by blocks and with constant values on each block.'
+      detail = 'with data sorted by blocks and with constant values on each block,'
       recover = false
       # Function used at nodes
       allow_warnings = true
     []
+    [multiple_files]
+      type = 'Exodiff'
+      input = 'piecewise_constant_elem_multiple.i'
+      exodiff = 'piecewise_constant_elem_multiple.e'
+      cli_args = 'Outputs/file_base=piecewise_constant_elem_multiple'
+      detail = 'with multiple CSV files read on every execution of the user object.'
+      # Cannot figure it out so it will error
+      recover = false
```

Comment:
still unsolved?

Reply:
yup. But maybe applying your patch it will resolve it. Let s see

Reply:
no it does not help. It s that unsigned int file index that does not get restarted somehow.
@loganharbour do you see anything wrong in how I declare / use `_current_file_index` ?